src/main/objects.c

```diff
@@ -991,16 +991,32 @@ static SEXP inherits3(SEXP x, SEXP what, SEXP which)
     else
 	PROTECT(klass = R_data_class(x, FALSE));
 
-    if(!isString(what))
-	error(_("'what' must be a character vector"));
+    int nprot = 1;
+
+	if (!isString(what))
+	{
+		if (inherits(what, "R7_class"))
+		{
+			SEXP R7_ns = R_FindNamespace(mkString("R7"));
+			SEXP cl = PROTECT(lang2(install("R7_class_name"), what));
+			what = PROTECT(eval(cl, R7_ns));
```

Comment:
Do we want to rely on an R7 R callback like this here?

Would it be an easier sell if this was "R7 agnostic" C code that just grabbed the `name` and `package` attributes off of the R7_class object and smushed them together separated by `::`?

I feel like R core is going to be hesitant to allow R's C code to call out to R code managed by a package on CRAN that in theory could be updated at any time to something incompatible with this

Reply:
If we do keep this, we could consider making `R7_ns` and `cl` static objects so we don't have to look them up every time (only once)

Reply:
This was discussed in the working group meeting--consensus was reached that accessing the S7 namespace is the preferred approach.
I updated the implementation by making the SEXPs static. Thank you for the suggestion!